Tallies/TallyClerks/MGxsClerk_class.f90

```diff
@@ -0,0 +1,887 @@
+  use scoreMemory_class,          only : scoreMemory
+  use tallyClerk_inter,           only : tallyClerk, kill_super => kill
+
+  implicit none
+  private
+
+  !! Size of clerk memory
+  integer(shortInt), parameter  :: ARRAY_SCORE_SIZE   = 7 ,&  ! Size of data to store as 1D arrays
+                                   MATRIX_SCORE_SMALL = 3 ,&  ! Size of data to store as 2D arrays when scoring up to P1
+                                   MATRIX_SCORE_FULL  = 9     ! Size of data to store as 2D arrays when scoring up to P7
+
+  !! Locations of different bins wrt memory address of the clerk
+  integer(longInt), parameter   :: FLUX_idx      = 1 ,&  ! Flux
+                                   SCATT_idx     = 2 ,&  ! Scattering macroscopic reaction rate
+                                   CAPT_idx      = 3 ,&  ! Capture macroscopic reaction rate
+                                   FISS_idx      = 4 ,&  ! Fission macroscopic reaction rate
+                                   NUBAR_idx     = 5 ,&  ! NuBar
+                                   CHI_idx       = 6 ,&  ! Fission neutron spectrum
+                                   SCATT_EV_idx  = 7     ! Analog: number of scattering events
+
+  !!
+  !! Multi-group macroscopic cross section calculation
+  !!
+  !! It prints out:
+  !! capture xs, fission xs, transport xs, nu, chi, the P0 and P1 scattering matrices,
+  !! the P0 scattering production matrix. On request, also the P2 -> P7 scattering matrices.
+  !!
+  !! NOTE:
+  !! - the cross sections are tallied with a collision estimator;
+  !! - the scattering matrices and chi are calculated with an analog estimator: their
+  !!   probability is calculated directly by scoring events
+  !! - the transport cross section is calculated with both the out-scatter and
+  !!   flux-limited approximation
+  !!
+  !! Private Members:
+  !!   energyMap -> tally map for energy group structure
+  !!   spacemap  -> tally map for material or spatial bins
+  !!   energyN   -> number of energy groups
+  !!   matN      -> number of materials or spatial bins
+  !!   width     -> number of memory bins needed
+  !!   PN        -> flag to indicate whether matrices above P1 are needed
+  !!
+  !! Interface
+  !!   tallyClerk Interface
+  !!
+  !! SAMPLE DICTIOANRY INPUT:
+  !!
+  !! myMGxsClerk {
+  !!   type MGxsClerk;
+  !!   # energyMap { energyMap definition } #
+  !!   # spaceMap  { <other tallyMap definition> } #
+  !!   # PN 1; #
+  !! }
+  !!
+  !!  NOTE: spaceMap can be any type of spatial tallyMap, including a multiMap
+  !!  NOTE: PN is a flag that indicates whether scattering matrices from P2 to P7
+  !!        have to be calculated
+  !!
+  type, public, extends(tallyClerk) :: MGxsClerk
+    private
+    ! Maps
+    class(tallyMap), allocatable :: spaceMap
+    class(tallyMap), allocatable :: energyMap
+
+    ! Useful data
+    integer(shortInt) :: energyN
+    integer(shortInt) :: matN
+    integer(shortInt) :: width
+    logical(defBool)  :: PN
```

Comment:
```suggestion
    integer(shortInt) :: energyN = 0 
    integer(shortInt) :: matN = 0
    integer(shortInt) :: width = 0
    logical(defBool)  :: PN = .false.
```
It is better to be explicit. For one I do not quite remember how Fortran default initialised bools ;-)